### ISSUE TITLE
Feat: Add exit confirmation dialog

### ISSUE THREAD

**Issue by htg2704** [opendatakit/skunkworks-crow#372]
#### Software and hardware versions 
All

#### Problem description
While at home screen, if back button is pressed, the app directly closes.
#### Steps to reproduce the problem
Open the app and click back.
#### Expected behavior
Upon adding the feature, a dialog box will be displayed which will ask user to confirm wether to close the app or back was pessed accidentlly.

**Comment by htg2704** [comment]
@opendatakit-bot claim --force

**Comment by Chromicle** [comment]
Hello, @htg2704,
A dialog should only be shown when there is a critical action that can't be undone easily eg logout, ending the transfer of forms, banking related things, etc.
we don't use any type of things in home screen so,  dialog to confirm if the user wants to exit is not necessary lets take @lakshyagupta21 and @huangyz0918 decisions also

**Comment by htg2704** [comment]
@Chromicle Even I thought the same but went ahead of opening the issue because of two reasons -
1.Due to different swipe gestures commonly used by various users, It may happen that a swipe by mistake may result in closing the app(Happened with me personally). It won't be a big problem but it results in bad user experience.
2.The app neither has any splash screen(saw the issue opened by fellow developer) nor any exit confirmation. It looks too plain.(Again not a problem, but diminishes user experience).
If the discussion leads to terming the dialog box unncessary then we can close the issue:)

**Comment by Chromicle** [comment]
> it may happen that a swipe by mistake may result in closing the app

That is the feature which will be in some of the devices only mostly android 9 and 10 and we don't have any right swipe features in main activity so, I think if and if a user is intended to exit app then only he will do that 

>  but diminishes the user experience

Can you explain a bit how it actually increases the user experience with the help of adding exit confirmation dialog

**Comment by htg2704** [comment]
Sorry for such a late reply, I was travelling due to current circumstances.
1. As most of the devices are either working on Andoid 9.0(+) or will receive updates in near future, I thought it is a valid reason. Speaking from the limited number of people I have seen using the swipe back button, this does irritate users.
2. As I said, right now our app don't have a splashscreen. Once it is included, closing the app by mistake and opening it with splash will again result in a bitter experience for user.

**Comment by pritsahkar** [comment]
Can we use "press again to exit" toast for it?

**Comment by htg2704** [comment]
@pritsahkar I am still assigned to the issue. You shouldn't have created without informing the orignal assignee.

**Comment by htg2704** [comment]
@Chromicle my pull request is addressing 2 issues, this one and adding back button in settings activity. But I think solution proposed by @pritsahkar is better and he has also linked a pull request.
Shall I revert the changes for this issue and create a new pull request for previous issue?

**Comment by Chromicle** [comment]
It fine if you make 2 pull requests addressing 2 issues, sorry for that I didn't checked you did in the same PR
@prajurock you don't have to create PR before claiming the issue 
@lakshyagupta21 @huangyz0918 can you please give your opinion about this issue, thanks

**Comment by huangyz0918** [comment]
> Can we use "press again to exit" toast for it? 

I agree with this one, since the `AlertDialog` is not a good option if user just want to quit from the APP's main page, as @Chromicle said that should be more suitable in stopping some time consuming tasks. 

> @Chromicle my pull request is addressing 2 issues, this one and adding back button in settings activity. But I think solution proposed by @pritsahkar is better and he has also linked a pull request.
Shall I revert the changes for this issue and create a new pull request for previous issue?

Sure! I think @pritsahkar 's PR is great and you can open another one or just remove the changes in the current one for another issue @htg2704 .

Thanks for all of your contributions, I'll take time to review each PR, but recently I'm busy with some COVID-19 things, sorry for the delay.

**Comment by pritsahkar** [comment]
@htg2704 @Chromicle @huangyz0918 apologize me, I am a beginner and I'm confused

**Comment by huangyz0918** [comment]
@pritsahkar It doesn't matter, all experts were beginners at first, just ask in the public channels like slack and forum if you need some help.
